Tasks/Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts

```diff
@@ -103,6 +76,33 @@ export class VstsServerUtils {
     }
 
     /**
+     * If enabled, fails the build in response to a quality gate failure.
+     * @param sqRunSettings SonarQubeRunSettings object for the applicable run or null if this is a PR build and the object could not be created
+     * @param sqMetrics     SonarQube metrics for the applicable run
+     * @returns {Promise<void>} Promise resolved when action completes (NB: Setting build result to failed results in procees exit)
+     */
+    private static breakBuildIfQualityGateFails(sqRunSettings:SonarQubeRunSettings, sqMetrics:SonarQubeMetrics):Q.Promise<void> {
```

Comment:
A decision was made to move the test lower down. This has the effect of reducing code coverage, but this is considered acceptable because the loss in code coverage is limited to a single line _tl.setResult()_, for which test coverage offers minimal value anyway.

There is also a beneficial side effect of being able to return the _breakBuildIfQualityGateFails()_ to private, when it was previously exposed for testing.